graph_weather/models/aurora/lora.py

```diff
@@ -0,0 +1,45 @@
+"""
+    Low-Rank Adaptation (LoRA) for Linear Layers:
+    - Adds trainable rank-constrained adaptations to frozen pre-trained models.
+"""
+import torch 
+import torch.nn as nn
+import math
+class LoraLayer(nn.Module):
```

Comment:
I'm not sure if it is directly comparable, but could you just import the Lora layer from Fengwu-GHR here? https://github.com/openclimatefix/graph_weather/blob/5908a56a3f5c1d08e62e4aa37f2b1bf05f72746f/graph_weather/models/fengwu_ghr/layers.py#L351 Or, if this implementation is better, could replace the Fengwu-GHR LoraLayer with this one? Just so we don't end up with multiple, slightly different LoRA layer implementations. This one seems to have more options, so I might recommend this one instead of the Fengwu one, but then it would be good to update the Fengu one. That could be a different PR though, to keep it smaller and easier to review.